crates/rslint_parser/src/syntax/jsx/mod.rs

```diff
@@ -68,13 +74,68 @@ pub(super) fn maybe_parse_jsx_expression(p: &mut Parser) -> ParsedSyntax {
 
 // test jsx jsx_element_as_statements
 // <div />
+
+// test_err jsx_or_type_assertion
+// function f() {
+//     let a = <div>a</div>; // JSX
+//     let b = <string>b; //type assertion
+//     let c = <string>b<a>d; // type assertion
+//     let d = <div>a</div>/; // ambigous: JSX or "type assertion a less than regex /div>/". Probably JSX.
+//     let d = <string>a</string>/;
+// }
 fn parse_jsx_expression(p: &mut CheckpointedParser<'_, '_>) -> ParsedSyntax {
     parse_jsx_element(p).map(|element| {
         let m = element.precede(p);
         m.complete(p, JsSyntaxKind::JSX_ELEMENT_EXPRESSION)
     })
 }
 
+struct JsxChildrenList;
+
+impl ParseNodeList for JsxChildrenList {
+    fn parse_element(&mut self, p: &mut Parser) -> ParsedSyntax {
+        // test jsx jsx_element_children
+        // <a>
+        //     <b>
+        //        <d></d>
+        //        <e></e>
+        //     </b>
+        //     <c></c>
+        // </a>
+        if p.at(T![<]) {
+            parse_jsx_element(p)
+        }
+        // test jsx jsx_expression_children
+        // <a>
+        //     {something}
+        //     {x => something}
+        // </a>
+        else if p.at(T!['{']) {
+            parse_jsx_expression_block(p, JsSyntaxKind::JSX_EXPRESSION_CHILD)
+        } else {
+            ParsedSyntax::Absent
+        }
+    }
+
+    fn is_at_list_end(&mut self, p: &mut Parser) -> bool {
+        let at_l_angle0 = p.at(T![<]);
+        let at_slash1 = p.nth_at(1, T![/]);
+        at_l_angle0 && at_slash1
+    }
+
+    fn recover(&mut self, p: &mut Parser, parsed_element: ParsedSyntax) -> RecoveryResult {
+        parsed_element.or_recover(
+            p,
+            &ParseRecovery::new(JsSyntaxKind::JS_UNKNOWN_MEMBER, token_set![T![<]]),
```

Comment:
Nit: We need to make sure that we don't eat over the element's closing token. 

```suggestion
            &ParseRecovery::new(JsSyntaxKind::JS_UNKNOWN_MEMBER, token_set![T![<], T![>], T!['{']]),
```

Reply:
Done.